### ISSUE TITLE
Wrong reading of X dataset from *.sm4 file containing STS data

### ISSUE THREAD

**Issue by andr-nau** [rescipy-project/spym#6]
**If I read .sm4 file that contains STS (scanning tunneling spectroscopy) data, the X dataset appears to have the same value for all items. It is equal to first X value from raw data.**

For example, if a raw data file have set of 100 (X,Y) items, and X changes from -0.5 to 0.5, then after reading this file, all X items in X dataset will have the same value -0.5. 
Tested for different raw *.sm4 files.

**Example with file provided:**
if I run the code:

```
f = spym.load('dI-dV_0767.sm4')
f.x
```

for this file - [dI-dV_0767.sm4 example file](https://aghedupl-my.sharepoint.com/:u:/g/personal/naumov_agh_edu_pl/EXCdPNmubFlGv-v38OFMG00B8pnmd-7gtYue-TtdrBj68g?e=gOkRdq)  (it consists of 1 X set and 12 Y sets, each with 551 elements. X changes from -0.5 to 0.6), 

as a result, Y sets will be loaded correctly, but X set - will be loaded with value -0.5 for all X. 

    xarray.DataArray
    'x'

    x: 551

    array([-0.5, -0.5, -0.5, ..., -0.5, -0.5, -0.5])

But it has to be from -0.5 to 0.6.

***

The **rhksm4** package reads such file in the right way for X.  The code

```
ds = rhksm4.to_dataset('dI-dV_0767.sm4')
ds.x
```

gives right result for X - it will be set of 551 values from -0.5 to 0.6 

    xarray.DataArray
    'x'

    x: 551

    array([-0.5  , -0.498, -0.496, ...,  0.596,  0.598,  0.6  ])

**Comment by andr-nau** [comment]
I don't see why it is so. Because it seems the spym uses the rhksm4 to read the data.
I'll do few more tests.

**Comment by mpanighel** [comment]
@andr-nau Thank you for you bug report!

The issue is with line [1177](https://github.com/rescipy-project/spym/blob/9200985aad8f6c89b52504555e6da406244493f6/spym/io/rhksm4/_sm4.py#L1177) in `spym/io/rhksm4/_sm4.py`

It should read
``` python
('x', xscale * np.arange(xsize, dtype=np.float64) + xoffset)
```
instead of 
``` python
('x', int(xscale) * np.arange(xsize, dtype=np.float64) + xoffset)
```

I will fix this in the next version. Let me know if in the meanwhile you could change it by your own and check that it works as expected for you.

Finally, spym does not have the rhksm4 package as dependency anymore. Its code is included in spym as a module.

**Comment by mpanighel** [comment]
Fixed in version 0.7.2 with commit https://github.com/rescipy-project/spym/commit/86fefb80ca7fb2db5f9e247c275b1065fc5d9c69

@andr-nau  Now you could also update the package through pip
